internal/internal_test.go

```diff
@@ -757,33 +760,84 @@ func TestGetResponseHttpStatus(t *testing.T) {
 	}
 }
 
-func TestGetParsedBodyJSON(t *testing.T) {
-	var req ext_authz.CheckRequest
-	if err := util.Unmarshal([]byte(exampleAllowedRequest), &req); err != nil {
-		panic(err)
-	}
-
-	result := getParsedBody(&req)
-
-	expected := map[string]interface{}{}
-	expected["firstname"] = "foo"
-	expected["lastname"] = "bar"
+func TestGetParsedBody(t *testing.T) {
 
-	if !reflect.DeepEqual(expected, result) {
-		t.Fatalf("Expected result %v but got %v", expected, result)
-	}
-}
+	requestNoContentType := `{
+		"attributes": {
+		  "request": {
+			"http": {
+			  "headers": {
+				"content-length": "0"
+			  }
+			}
+		  }
+		}
+	  }`
+
+	requestContentTypeText := `{
+		"attributes": {
+		  "request": {
+			"http": {
+			  "headers": {
+				"content-type": "text/html"
+			  }
+			}
+		  }
+		}
+	  }`
+
+	requestContentTypeJSON := `{
+		"attributes": {
+		  "request": {
+			"http": {
+			  "headers": {
+				"content-type": "application/json"
+			  },
+			  "body": "{\"firstname\": \"foo\", \"lastname\": \"bar\"}"
+			}
+		  }
+		}
+	  }`
+
+	tests := map[string]struct {
+		input *ext_authz.CheckRequest
+		want  int
```

Comment:
Perhaps we should just assert on the body content here? It's not clear why we would want to assert on the number of elements.

Reply:
Updated test to assert body.